resources/servicecatalog-portfolio-share-attachements.go

```diff
@@ -0,0 +1,90 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+)
+
+type ServiceCatalogPortfolioShareAttachment struct {
+	svc         *servicecatalog.ServiceCatalog
+	portfolioID *string
+	accountID   *string
+}
+
+func init() {
+	register("ServiceCatalogPortfolioShareAttachment", ListServiceCatalogPortfolioShareAttachments)
+}
+
+func ListServiceCatalogPortfolioShareAttachments(sess *session.Session) ([]Resource, error) {
+	svc := servicecatalog.New(sess)
+	resources := []Resource{}
+	portfolios := []*servicecatalog.PortfolioDetail{}
+	accounts := []*string{}
+
+	params := &servicecatalog.ListPortfoliosInput{
+		PageSize: aws.Int64(20),
+	}
+
+	//List all Portfolios
+	for {
+		resp, err := svc.ListPortfolios(params)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, portfolioDetail := range resp.PortfolioDetails {
+			portfolios = append(portfolios, portfolioDetail)
+		}
+
+		if resp.NextPageToken == nil {
+			break
+		}
+
+		params.PageToken = resp.NextPageToken
+	}
+
+	accessParams := &servicecatalog.ListPortfolioAccessInput{}
+
+	// Get all accounts which have shared access to the portfolio
+	for _, portfolio := range portfolios {
+
+		accessParams.PortfolioId = portfolio.Id
+
+		resp, err := svc.ListPortfolioAccess(accessParams)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, accountID := range resp.AccountIds {
+			accounts = append(accounts, accountID)
+		}
+
+		for _, account := range accounts {
```

Comment:
I would suggest dropping the `accounts` slice and directly iterating over `resp.AccountIds`.